sdcm/cluster.py

```diff
@@ -2005,66 +2019,82 @@ def download_scylla_repo(self, scylla_repo):
-        self.remoter.sudo(f'{package_mgr} install -y {package_name}')
+        self.remoter.sudo(f'{"yum" if self.distro.is_rhel_like else "apt-get"} install -y {package_name}')
 
-    def is_apt_lock_free(self):
-        cmd = 'lsof /var/lib/dpkg/lock'
-        res = self.remoter.sudo(cmd, ignore_status=True)
-        if res.exit_status == 0:
-            return True
-        else:
-            return False
+    def is_apt_lock_free(self) -> bool:
+        return self.remoter.sudo("lsof /var/lib/dpkg/lock", ignore_status=True).ok
 
-    def install_manager_agent(self, package_path=None):
-        auth_token = Setup.test_id()
-        manager_prometheus_port = self.parent_cluster.params.get("manager_prometheus_port")
+    def install_manager_agent(self, package_path: Optional[str] = None) -> None:
         if package_path:
-            package_name = '{}scylla-manager-agent*'.format(package_path)
+            package_name = f"{package_path}scylla-manager-agent*"
         else:
             self.download_scylla_manager_repo(
                 self.parent_cluster.params.get("scylla_mgmt_agent_repo") or
                 self.parent_cluster.params.get("scylla_mgmt_repo"))
-            package_name = 'scylla-manager-agent'
-        package_mgr = "yum" if self.distro.is_rhel_like else "apt-get"
-        self.install_package(package_mgr, package_name)
-        install_and_config_agent_command = dedent(r"""
-            sed -i 's/# auth_token:.*$/auth_token: {}/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            sed -i 's/#tls_cert_file/tls_cert_file/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            sed -i 's/#tls_key_file/tls_key_file/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            sed -i 's/#prometheus: .*/prometheus: :{}/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
+            package_name = "scylla-manager-agent"
+
+        self.install_package(package_name)
+
+        with self.remote_manager_agent_yaml() as manager_agent_yaml:
+            manager_agent_yaml["auth_token"] = Setup.test_id()
+            manager_agent_yaml["tls_cert_file"] = None
+            manager_agent_yaml["tls_key_file"] = None
+            manager_agent_yaml["prometheus"] = f":{self.parent_cluster.params.get('manager_prometheus_port')}"
+
+        self.remoter.sudo(shell_script_cmd("""\
             systemctl restart scylla-manager-agent
             systemctl enable scylla-manager-agent
-        """.format(auth_token, manager_prometheus_port))
-        self.remoter.run('sudo bash -cxe "%s"' % install_and_config_agent_command)
-        version = self.remoter.run('scylla-manager-agent --version').stdout
-        self.log.info(f'node {self.name} has scylla-manager-agent version {version}')
+        """))
+
+        manager_agent_version = self.remoter.run("scylla-manager-agent --version").stdout
+        self.log.info("node %s has scylla-manager-agent version %s", self.name, manager_agent_version)
```

Comment:
It's called lazy logging. And this is preferred way to format log messages in Python.

There are several pylint warnings about that (e.g., W1201 and W1202)

fyi, I just found a short blog post on medium.com: https://medium.com/swlh/why-it-matters-how-you-log-in-python-1a1085851205

Reply:
@enaydanov, actually this  is not that true anymore.
It was being discussed in the  community  for quite long  time  (see for example https://github.com/PyCQA/pylint/issues/2395 and  its realted issues), but eventually in pylint 2.4 they allowed the  options to use f-string in logging  (which I prefer in most our cases): https://github.com/PyCQA/pylint/commit/eeca6cf0baa7263ff46598f06e4eda0f3072262b
http://pylint.pycqa.org/en/latest/whatsnew/2.4.html?highlight=fstring%20interpolation

Reply:
@roydahan log string formatting style and lazy logging are orthogonal things.

A formatting style is an option for the logging stuff: https://docs.python.org/3/howto/logging-cookbook.html#using-particular-formatting-styles-throughout-your-application

And the lazy logging is about a time when string bits composed together. And this thing still true.

Also, from the doc you provided the link to:
```
...
- Allow the option of f-strings as a valid logging string formatting method.

logging-fstring-interpolation has been merged into logging-format-interpolation to allow the logging-format-style option to control which logging string format style is valid. To allow this, a new fstr value is valid for the logging-format-style option.
...
```

Note, that a valid format style controlled by `logging-format-style` pylint' option.

Reply:
I'm not sure I'm following you.
I think the original comment was why you change from f-string format to %-format.
f-string format should be fine.

Reply:
Again, the change is not about formatting but about lazy logging. To make the common ground I need to ask you to provide an example how to use f-string in combination with lazy logging. This will clear all questions.